### PR TITLE
Address https://github.com/dotnet/xharness/issues/177

### PR DESCRIPTION
**Pull request by MattGal**
Still testing this out but this should solve https://github.com/dotnet/xharness/issues/177 and make it obvious to users when things crash.

### CODE REVIEW

**Comment by MattGal** [comment]
image

File: src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs [hunk 1]
```diff
@@ -10,23 +10,38 @@
 using Microsoft.DotNet.XHarness.Android;
 using Microsoft.DotNet.XHarness.CLI.CommandArguments;
 using Microsoft.DotNet.XHarness.CLI.CommandArguments.Android;
+using Microsoft.DotNet.XHarness.iOS.Shared.Logging;
```
Comment (mandel-macaque) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs:13]: @MattGal looks like the IDE added a namespace you dont need?

File: src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs [hunk 2]
```diff
@@ -117,6 +132,27 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
                         {
                             logger.LogInformation($"Test execution summary:{Environment.NewLine}{resultValues[_testRunSummaryVariableName]}");
                         }
+
+                        // Due to the particulars of how instrumentations work, ADB will report a 0 exit code for crashed instrumentations
+                        // We'll change that to a specific value and print a message explaining why.
+                        if (resultValues.ContainsKey(_returnCodeVariableName))
```
Comment (premun) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs:138]: nit: you could also `TryGetValue` and use it below at :140 and :147

Comment (MattGal) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs:139]: IMO that's a personal preference, doesn't really simplify or make more obvious, so I'm going to skip this.

**Pull request merged**